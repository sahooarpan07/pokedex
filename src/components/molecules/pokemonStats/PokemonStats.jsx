import {View, Text} from 'react-native';
import React from 'react';
import ProgressBar from '../../atoms/progressBar/ProgressBar';
import styles from './PokemonStats.styles';
const PokemonStats = ({stats = []}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>Stats</Text>
      <View style={styles.wrapper}>
        {stats.map(data => {
          return (
            <ProgressBar
              label={data?.stat?.name.toUpperCase()}
              value={data.base_stat / 100}
            />
          );
        })}
      </View>
    </View>
  );
};

export default PokemonStats;
