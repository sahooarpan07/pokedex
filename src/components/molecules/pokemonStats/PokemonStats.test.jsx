/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonStats from './PokemonStats';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders Pokemon Stats correctly', () => {
  const stats = [
    {
      stat: {
        name: 'Attack',
      },
      base_stat: 74,
    },
  ];

  const tree = renderer.create(<PokemonStats stats={stats} />).toJSON();

  expect(tree).toMatchSnapshot();
});
