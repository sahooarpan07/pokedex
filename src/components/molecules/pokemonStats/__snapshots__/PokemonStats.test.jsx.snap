// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon Stats correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#65bfb6",
      "marginVertical": 8,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#180345",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    Stats
  </Text>
  <View
    style={
      {
        "flexDirection": "column",
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "color": "#180345",
            "fontSize": 18,
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
        ATTACK
      </Text>
      <View
        style={
          {
            "margin": 8,
          }
        }
      >
        <View
          animated={true}
          animationConfig={
            {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          indeterminateAnimationDuration={1000}
          onLayout={[Function]}
          progress={0.74}
          style={
            [
              {
                "backgroundColor": undefined,
                "borderColor": "#180345",
                "borderRadius": 4,
                "borderWidth": 1,
                "overflow": "hidden",
                "width": 200,
              },
              {
                "position": "relative",
              },
            ]
          }
          useNativeDriver={false}
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#180345",
                "height": 12,
                "transform": [
                  {
                    "translateX": -7.105427357601002e-15,
                  },
                  {
                    "translateX": -25.739999999999995,
                  },
                  {
                    "scaleX": 0.740026,
                  },
                ],
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "bold",
            "left": 200,
            "position": "absolute",
            "top": 4,
          }
        }
      >
        74
      </Text>
    </View>
  </View>
</View>
`;
